lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1630,6 +1630,9 @@ bool SessionState::receive_charging_credit(
                 << " Activating service RG: " << key << " for " << session_id_;
     grant->set_service_state(SERVICE_NEEDS_ACTIVATION, credit_uc);
   }
+  if (grant->should_deactivate_service()) {
+    grant->set_service_state(SERVICE_NEEDS_DEACTIVATION, credit_uc);
+  }
```

Comment:
why do we check it here. Don't we have to wait for the next cycle to check?

Reply:
@uri200 This case will only cover the case where the credit is already exhausted and FUA-terminate is set. Since at this point no new usage is added, it just acts on the terminate instruction. 
(I don't think it makes much sense to wait until the next cycle in this case)